### PR TITLE
fix: remove upperbound value if max field is 0

### PR DESCRIPTION
**Pull request by Silvia-loza**
Related Issue:  https://oat-sa.atlassian.net/browse/AUT-12

Select point interaction response Scoring defaults to 0 even if correct when tickbox is set to 0.
IU allows the score max to go to 0 even when the box is ticked.

FIXES:
Scoring issue => conditional in formElement changed to add value 0 to the conditions that  remove upperbound.
IU issue = > Removing a threshold set to 0 in minMaxComponentFactory.

SETPS to test:
1. Create a new item
2. Add a Select Point interaction into it
3. Define a response area with score=1
4. Define max score range value
5. Save the item (you may find a QTI sample of the state below  )
6. When maximum Score checked try to go below one.
8. Undefine max score range value by unchecking the corresponding checkbox
9. Save the item (you may find a QTI sample of the state below  )
10. Preview the item
11. Select the correct response
12. Submit the response
13. Response should contain the right score.

Actual result: incrementor allows 0 even with the box ticked, Score is 0 even if right answer in that case.
Expected result: incrementor doesn't allow going down from one if box ticked, result is always given points when answer is correct.

### CODE REVIEW

**Review by bziondik — APPROVED** [review]
- [ ]  New code is covered by tests (if applicable)
- [x]  Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x]  New code is respecting code style rules
- [x]  New code is respecting best practices
- [x]  New code is not subject to concurrency issues (if applicable)
- [x]  Feature is working correctly on my local machine (if applicable)
- [x]  Acceptance criteria are respected
- [x]  Pull request title and description are meaningful

**Review by AnastasiaArcadia — CHANGES_REQUESTED** [review]
Unfortunately, this affects another interactions. For example for the Associate Interaction I'm getting an error now:
Снимок экрана 2021-07-01 в 12 52 15

File: views/js/qtiCreator/widgets/helpers/formElement.js [hunk 1]
```diff
@@ -288,15 +291,15 @@ define([
                 if (name === 'upperBound' && this.disabled) {
                     // if the field is disabled, the corresponding attribute should be removed.
                     element[options.attrMethodNames.remove](name);
-                } else if (!value && (element.is('orderInteraction') || element.is('graphicOrderInteraction'))) {
-                    element[options.attrMethodNames.remove](name); //to be removed for order interactions
-                } else {
+                } else if (!options.allowNull && (value === 0 || !isActualNumber)) {
+                    //if a null attribute is not allowed, should be removed.
```
[the remark this answers is not in the document]
Comment (Silvia-loza) [reply views/js/qtiCreator/widgets/helpers/formElement.js]: All this part has been removed at the end.

**Review by AnastasiaArcadia — APPROVED** [review]
- [ ]  New code is covered by tests (if applicable)
- [x]  Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x]  New code is respecting code style rules
- [x]  New code is respecting best practices
- [x]  New code is not subject to concurrency issues (if applicable)
- [x]  Feature is working correctly on my local machine (if applicable)
- [x]  Acceptance criteria are respected
- [x]  Pull request title and description are meaningful

**Comment by AntonNeviarouski** [comment]
QA verified on http://test-silvia.playground.kitchen.it.taocloud.org:41441/

**Pull request merged**